### ISSUE TITLE
problem with sending emails

### ISSUE THREAD

**Issue by klausmcm** [ampache/ampache#2342]
## Description

I have problems sending an email as admin from ampache (4.1.1). I'm using dismail as the email provider (https://dismail.de/info.html).

### Describe the bug

Using *E-Mail Users* as admin

![image](https://user-images.githubusercontent.com/14914359/78213239-f12d8580-7466-11ea-93a9-977c1851da62.png)

leads to this

![image](https://user-images.githubusercontent.com/14914359/78213256-ff7ba180-7466-11ea-8b6f-40ac83588c3c.png)

As reference, I have email sending working properly in **thunderbird**

![image](https://user-images.githubusercontent.com/14914359/78213468-9c3e3f00-7467-11ea-9881-04ec059cceb7.png)

and self-hosted **gitlab**

![image](https://user-images.githubusercontent.com/14914359/78213613-1242a600-7468-11ea-85fc-ff26aae282d8.png)

I have tried different combinations of ports for *mail_port* and values for *mail_secure_smtp* without success. I have tried enabling debugging in ampache (level 5) but failed email sending don't seem to show up in the log.

### Expected behavior

Probably some screen that says that the email was sent successfully and an email in my regular account (configured with a working email address).

## Environment

* Ampache version: 4.1.1
* Web server + version: 2.4.38
* Server operating system: Debian 10
* Client operating system: Debian 9

### Client type

* Web

## Settings

Configuration in *ampache.cfg.php*

```
; Enable or disable email server features
; otherwise, you can reset your password
; and never receive an email with the new one
; Default: false
mail_enable = "true"

; Method used to send mail
; POSSIBLE VALUES: smtp sendmail php
; DEFAULT: php
mail_type = "smtp"

; Mail domain.
; DEFAULT: example.com
mail_domain = "dismail.de"

; This will be combined with mail_domain and used as the source address for
; emails generated by Ampache.  For example, setting this to 'me' will set the
; sender to 'me@example.com'.
; DEFAULT: info
mail_user = "emailuser"

; A name to go with the email address.
; DEFAULT: Ampache
mail_name = "AmpacheMail"

; How strictly email addresses should be checked.
; easy does a regex match, strict actually performs some SMTP transactions
; to see if we can send to this address.
; POSSIBLE VALUES: strict easy none
; DEFAULT: strict
mail_check = "easy"

; Mail server (hostname or IP address)
; DEFAULT: localhost
mail_host = "smtp.dismail.de"

; SMTP port
; DEFAULT: 25
mail_port = 465

; Secure SMTP
; POSSIBLE VALUES: ssl tls
; DEFAULT: none
mail_secure_smtp = ssl

; Enable SMTP authentication
; DEFAULT: false
mail_auth = "true"

; SMTP username
; your mail auth username.
mail_auth_user = "emailuser@dismail.de"

; SMTP password
; your mail auth password.
mail_auth_pass = "---"
```

**Comment by lachlan-00** [comment]
@klausmcm the mail server settings won't change but the mail_type tells php which settings to use. smtp is a direct connection in php so it would definitely log any errors you get. i set up sendmail on the server so i didn't have to configure those things.

![image](https://user-images.githubusercontent.com/1305249/78217771-030f3700-7500-11ea-9630-f796c56d5200.png)

**Comment by klausmcm** [comment]
That's good to know. Would you please review my mail related configuration?

Based on the values provided by the email provider

![image](https://user-images.githubusercontent.com/14914359/78218735-2049f400-7473-11ea-8e90-0bf681c23431.png)

would you expect this configuration to work? Any flaws? I have the same account working fine in thunderbird and gitlab so I'm not sure if I'm missing something simple.

```
mail_enable = "true"
mail_type = "smtp"
mail_domain = "dismail.de"
mail_user = "emailuser"
mail_name = "AmpacheMail"
mail_check = "easy"
mail_host = "smtp.dismail.de"
mail_port = 465
mail_secure_smtp = ssl
mail_auth = "true"
mail_auth_user = "emailuser@dismail.de"
mail_auth_pass = "---"
```

**Comment by lachlan-00** [comment]
those settings seem correct, ssl 465. and the fully qualified user for mailuser.

you should tail -f the ampache log and the php log to check for errors, email gets sent when pm'ing, shouts and a few other things
(these are where my logs are)
```
tail -f /var/log/ampache/ampache.log
tail -f /var/log/apache2/error.log
```

Now that we've been talking about it i need to put more debuggin into the ampache log so i'll do that now.

**Comment by klausmcm** [comment]
I still have the issue. This is the config I used for testing

```
debug = "true"
debug_level = 5
log_path = "/var/log/ampache"
log_filename = "%name.%Y%m%d.log"

mail_enable = "true"
mail_type = "smtp"
mail_domain = "dismail.de"
mail_user = "emailuser"
mail_name = "AmpacheMail"
mail_check = "easy"

mail_host = "smtp.dismail.de"
mail_port = 465
mail_secure_smtp = ssl
mail_auth = "true"
mail_auth_user = "emailuser@dismail.de"
mail_auth_pass = "password"
```

When I then try to send an email

![image](https://user-images.githubusercontent.com/14914359/78328098-315a3a00-7533-11ea-8c4f-898a70cb7259.png)

I get this in the ampache log (after restarting apache and accessing ampache in a new browser session). I don't see any obvious issues related to email though.

```
2020-04-02 22:43:02 [ampache] (access.class) -> check_network interface: ip matched ACL 10.123.0.1
2020-04-02 22:43:02 [admin] (session.class) -> Session created: ff72mnigot1f5j03qnl9dqeb0p
2020-04-02 22:43:02 [ampache] (session.class) -> Existing session found
2020-04-02 22:43:02 [admin] (session.class) -> ff72mnigot1f5j03qnl9dqeb0p has been extended to Thu, 02 Apr 2020 23:43:02 -0700 extension length 3600
2020-04-02 22:43:02 [admin] (session.class) -> Session created: b57e9eb98eb9d8ad42077800c3d821c1
2020-04-02 22:43:03 [ampache] (session.class) -> Existing session found
2020-04-02 22:43:03 [admin] (session.class) -> ff72mnigot1f5j03qnl9dqeb0p has been extended to Thu, 02 Apr 2020 23:43:03 -0700 extension length 3600
2020-04-02 22:43:03 [ampache] (session.class) -> Existing session found
2020-04-02 22:43:03 [admin] (session.class) -> ff72mnigot1f5j03qnl9dqeb0p has been extended to Thu, 02 Apr 2020 23:43:03 -0700 extension length 3600
2020-04-02 22:43:03 [admin] (ajax.server) -> Called for page: {index}
2020-04-02 22:43:10 [ampache] (session.class) -> Existing session found
2020-04-02 22:43:10 [admin] (session.class) -> ff72mnigot1f5j03qnl9dqeb0p has been extended to Thu, 02 Apr 2020 23:43:10 -0700 extension length 3600
2020-04-02 22:43:10 [admin] (ajax.server) -> Called for page: {index}
2020-04-02 22:43:12 [ampache] (session.class) -> Existing session found
2020-04-02 22:43:12 [admin] (session.class) -> ff72mnigot1f5j03qnl9dqeb0p has been extended to Thu, 02 Apr 2020 23:43:12 -0700 extension length 3600
2020-04-02 22:43:12 [admin] (session.class) -> Session created: f5f2789c70c8054f8267d2fff569b351
2020-04-02 22:43:30 [ampache] (session.class) -> Existing session found
2020-04-02 22:43:30 [admin] (session.class) -> ff72mnigot1f5j03qnl9dqeb0p has been extended to Thu, 02 Apr 2020 23:43:30 -0700 extension length 3600
2020-04-02 22:43:30 [admin] (session.class) -> Session created: d52ce676ec97a2063c578d5c8c77d3df
2020-04-02 22:43:31 [admin] (core.class) -> Registered post form confirmation with SID 87b3edd762fbce1f5bdcc5d649d8539e and expiration 1585896211 (3600 seconds from now)
```

I have tried another email provider but I can't tell from the logs where the issue is.

**Comment by lachlan-00** [comment]
I think this is resolved in develop in 21f049a0e
![image](https://user-images.githubusercontent.com/1305249/99352368-3a472280-28ee-11eb-89d1-440a999f4f25.png)

**Comment by klausmcm** [comment]
@lachlan-00 Thank you. I hope to review this when the new release comes out.
